server/src/main/java/org/elasticsearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -125,18 +124,19 @@ abstract class AbstractSearchAsyncAction<Result extends SearchPhaseResult> exten
     ) {
         super(name);
         this.namedWriteableRegistry = namedWriteableRegistry;
-        final List<SearchShardIterator> toSkipIterators = new ArrayList<>();
         final List<SearchShardIterator> iterators = new ArrayList<>();
+        int skipped = 0;
         for (final SearchShardIterator iterator : shardsIts) {
             if (iterator.skip()) {
-                toSkipIterators.add(iterator);
+                skipped++;
             } else {
                 iterators.add(iterator);
             }
         }
-        this.toSkipShardsIts = toSkipIterators;
+        this.skippedCount = skipped;
         this.shardsIts = iterators;
-        outstandingShards = new AtomicInteger(shardsIts.size());
+        outstandingShards = new AtomicInteger(shardsIts.size() - skipped);
```

Comment:
Here it's slightly confusing which `shardsIts` it is - the parameter `shardsIts` as I see contains the skipped ones, but `this.shardsIts` does not. Maybe use a different name for the parameter? Also, isn't it `iterators.size()`?

Reply:
To think about it, do we really need `iterators` at all - can't we just work on `this.shardsIts` directly?

Reply:
It's just a little faster to have a dedicated local variable instead of looking up the field repeatedly, that's all :) There's no other reason to have a separate local variable.

But you're right the naming is horrific :O lets fix that in a follow-up but at least use `iterators.size()` here :) My idea/hope was actually to get rid of `this.shardIts`, it's somewhat unnecessary in the grand scheme of things :)